### PR TITLE
Cuong/feature/auth implement

### PR DESCRIPTION
**Pull request by ntc2019**
First commit . Implemented auth feature.

### CODE REVIEW

File: src/axios/Auth/instance.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+})
+
+export default authAxios
+
+authAxios.interceptors.request.use((config) => {
+    config.headers = {
+        ...config.headers,
+        'Authorization': localStorage.getItem('accessToken')
+            ? localStorage.getItem('accessToken')
+            : ''
```
Comment (Khanhnqps10920) [inline src/axios/Auth/instance.ts]:
```suggestion
        'Authorization': localStorage.getItem('accessToken')
```

**Comment by Khanhnqps10920** [comment]
Please take me a record video to show the login function

File: src/contexts/AuthProvider.tsx [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
-    // fake data
-
-    // if want login just return fake data
-    const fakeData = {
-      id: '1',
-      email: '222',
-    };
-
-    // return fakeData;
+    if (localStorage.getItem('accessToken')) {
```
Comment (Khanhnqps10920) [inline src/contexts/AuthProvider.tsx]: đoạn này ko cần check if

File: src/contexts/AuthProvider.tsx [hunk 3]
```diff
@@ -1,28 +1,47 @@
...
 export const authProvider: AuthProvider = {
   login: async (username: string, password: string): Promise<void> => {
     // call api login
+    try {
+      const userInfo = {
+        username,
+        password,
+      }
+      const response = await authAxios.post('/auth/login', userInfo);
+      const accessToken = 'Bearer ' + response.data['access_token']
```
Comment (Khanhnqps10920) [inline src/contexts/AuthProvider.tsx]: đoạn này lưu token vào localstorage, còn bearer sẽ được gắn bên axios config

**Comment by ntc2019** [comment]
chrome_cfb3t87mLT
Login Preview.

**Comment by Khanhnqps10920** [comment]
> chrome_cfb3t87mLT [ chrome_cfb3t87mLT ](https://user-images.githubusercontent.com/28971943/209548768-0c1cd393-5e3e-4898-b753-247b6f96c635.gif) [ ](https://user-images.githubusercontent.com/28971943/209548768-0c1cd393-5e3e-4898-b753-247b6f96c635.gif) Login Preview.

disable button when waiting for api response

**Comment by ntc2019** [comment]
chrome_tbHs43QdAU
Implement disabled button while waiting for api response.

**Pull request merged**